packages/salesforce-adapter/src/filters/currency_iso_code.ts

```diff
@@ -0,0 +1,114 @@
+/*
+*                      Copyright 2022 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import _ from 'lodash'
+import { Element, isObjectType, ObjectType, ElemID, ListType, InstanceElement, CORE_ANNOTATIONS, ReferenceExpression } from '@salto-io/adapter-api'
+import Joi from 'joi'
+import { FilterWith } from '../filter'
+import { SALESFORCE, FIELD_ANNOTATIONS, RECORDS_PATH, SETTINGS_PATH, CUSTOM_VALUE, CURRENCY_CODE_TYPE_NAME } from '../constants'
+import { Types, getTypePath } from '../transformers/transformer'
+
+const currencyCodeType = new ObjectType(
+  {
+    elemID: new ElemID(SALESFORCE, CURRENCY_CODE_TYPE_NAME),
+    fields: {
+      [FIELD_ANNOTATIONS.VALUE_SET]: { refType: new ListType(Types.valueSetType) },
+    },
+    annotations: {
+      [CORE_ANNOTATIONS.CREATABLE]: false,
+      [CORE_ANNOTATIONS.DELETABLE]: false,
+      [CORE_ANNOTATIONS.UPDATABLE]: false,
+    },
+    isSettings: true,
+    path: getTypePath(CURRENCY_CODE_TYPE_NAME),
+  }
+)
+
+const CURRENCY_CODE_FIELD_NAME = 'CurrencyIsoCode'
+
+type ValueSet = {}
+
+type CurrencyIsoCodeType = ObjectType & {
+  fields: {
+    [CURRENCY_CODE_FIELD_NAME]: {
+      annotations: {
+        valueSet: ValueSet[]
+        valueSetName?: ReferenceExpression
+      }
+    }
+  }
+}
+
+const VALUE_SET_SCHEMA = Joi.object({
+  valueSet: Joi.array().items({
+    [CUSTOM_VALUE.FULL_NAME]: Joi.string().required(),
+    [CUSTOM_VALUE.DEFAULT]: Joi.boolean().required(),
+    [CUSTOM_VALUE.LABEL]: Joi.string().required(),
+    [CUSTOM_VALUE.IS_ACTIVE]: Joi.boolean().required(),
+  }).required(),
+}).unknown(true).required()
+
+const isTypeWithCurrencyIsoCode = (elem: ObjectType): elem is CurrencyIsoCodeType => {
+  if (!_.has(elem.fields, CURRENCY_CODE_FIELD_NAME)) {
```

Comment:
nit: when using `lodash` has no significant readability benefit, we prefer to avoid it.
```suggestion
  if (elem.fields[CURRENCY_CODE_FIELD_NAME] === undefined) {
```
or
```suggestion
  if (!CURRENCY_CODE_FIELD_NAME in elem.fields) {
```
or
```suggestion
  if (!Object.prototype.hasOwnProperty(elem.fields, CURRENCY_CODE_FIELD_NAME)) {
```